### ISSUE TITLE
CacheMissException when running cache only

### ISSUE THREAD

**Issue by b99andla** [zino-hofmann/graphql-flutter#779]
**Describe the issue**

If i enable cache and i switch off the network after fetching the data and try to fetch the data again i get CacheMissException instead of the cached response

**To Reproduce**

    Set fetchPolicy: FetchPolicy.cacheAndNetwork,
    Call a graphql Query
    After that change the fetch policy to fetchPolicy: FetchPolicy.cacheOnly,
    You will see the error

**Expected behavior**
If there is no error getting the data using only cache should work.

**Message**
An exception containing  link exception "Round trip cache re-read failed: cache.readQuery(request) returned null"

**device / execution context**
IOS Simulator / Android simulator debbuing via VS code.

**Query**
```graphql
query Events {
  user {
    events {
      id
      time
      title
      text
      __typename
      receptacle {
        wasteType
        pickupLocation {
          street
          streetNr
          city
        }
      }
    }
  }
}
```

**Comment by micimize** [comment]
@b99andla can you provide the code you're using, and `print(result)` or `print([result.source, result.exception])` for each change?

It's hard to think of what could be causing this issue, so as many details as possible will help

**Comment by micimize** [comment]
Unable to reproduce so far. My guess is that somehow the cache is being cleared or emptied – will need to know exactly what happens between the `Query` calls – hot reloading or logic-driven, etc, as well as any other potential confounding variables.

**Comment by aseef17** [comment]
> Unable to reproduce so far. My guess is that somehow the cache is being cleared or emptied – will need to know exactly what happens between the `Query` calls – hot reloading or logic-driven, etc, as well as any other potential confounding variables.

Hey, I've been able to produce this too.
I assume this is happening cause i get the rawHTML in a variable.

The HTML looks something like

```
<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<link rel="preconnect" href="https://fonts.gstatic.com">
<link href="https://fonts.googleapis.com/css2?family=Noto+Sans+JP:wght@100;300;400;500;700;900&display=swap" rel="stylesheet">
<title>Inline css</title>
 <style>
    /*
 * CKEditor 5 (v24.0.0) content styles.
 * Generated on Thu, 10 Dec 2020 08:15:26 GMT.
 * For more information, check out https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/content-styles.html
 */

:root {
    --ck-color-mention-background: hsla(341, 100%, 30%, 0.1);
    --ck-color-mention-text: hsl(341, 100%, 30%);
    --ck-highlight-marker-blue: hsl(201, 97%, 72%);
    --ck-highlight-marker-green: hsl(120, 93%, 68%);
    --ck-highlight-marker-pink: hsl(345, 96%, 73%);
    --ck-highlight-marker-yellow: hsl(60, 97%, 73%);
    --ck-highlight-pen-green: hsl(112, 100%, 27%);
    --ck-highlight-pen-red: hsl(0, 85%, 49%);
    --ck-image-style-spacing: 1.5em;
    --ck-todo-list-checkmark-size: 16px;
}

/* ckeditor5-image/theme/imageresize.css */
.ck-content .image.image_resized {
    max-width: 100%;
    display: block;
    box-sizing: border-box;
}
/* ckeditor5-image/theme/imageresize.css */
.ck-content .image.image_resized img {
    width: 100%;
}
/* ckeditor5-image/theme/imageresize.css */
.ck-content .image.image_resized > figcaption {
    display: block;
}
/* ckeditor5-image/theme/imagestyle.css */
.ck-content .image-style-side {
    float: right;
    margin-left: var(--ck-image-style-spacing);
    max-width: 50%;
}
/* ckeditor5-image/theme/imagestyle.css */
.ck-content .image-style-align-left {
    float: left;
    margin-right: var(--ck-image-style-spacing);
}
/* ckeditor5-image/theme/imagestyle.css */
.ck-content .image-style-align-center {
    margin-left: auto;
    margin-right: auto;
}
/* ckeditor5-image/theme/imagestyle.css */
.ck-content .image-style-align-right {
    float: right;
    margin-left: var(--ck-image-style-spacing);
}
/* ckeditor5-image/theme/image.css */
.ck-content .image {
    display: table;
    clear: both;
    text-align: center;
    margin: 1em auto;
}
/* ckeditor5-image/theme/image.css */
.ck-content .image img {
    display: block;
    margin: 0 auto;
    max-width: 100%;
    min-width: 50px;
}
/* ckeditor5-image/theme/imagecaption.css */
.ck-content .image > figcaption {
    display: table-caption;
    caption-side: bottom;
    word-break: break-word;
    color: hsl(0, 0%, 20%);
    background-color: hsl(0, 0%, 97%);
    padding: .6em;
    font-size: .75em;
    outline-offset: -1px;
}
/* ckeditor5-highlight/theme/highlight.css */
.ck-content .marker-yellow {
    background-color: var(--ck-highlight-marker-yellow);
}
/* ckeditor5-highlight/theme/highlight.css */
.ck-content .marker-green {
    background-color: var(--ck-highlight-marker-green);
}
/* ckeditor5-highlight/theme/highlight.css */
.ck-content .marker-pink {
    background-color: var(--ck-highlight-marker-pink);
}
/* ckeditor5-highlight/theme/highlight.css */
.ck-content .marker-blue {
    background-color: var(--ck-highlight-marker-blue);
}
/* ckeditor5-highlight/theme/highlight.css */
.ck-content .pen-red {
    color: var(--ck-highlight-pen-red);
    background-color: transparent;
}
/* ckeditor5-highlight/theme/highlight.css */
.ck-content .pen-green {
    color: var(--ck-highlight-pen-green);
    background-color: transparent;
}
/* ckeditor5-font/theme/fontsize.css */
.ck-content .text-tiny {
    font-size: .7em;
}
/* ckeditor5-font/theme/fontsize.css */
.ck-content .text-small {
    font-size: .85em;
}
/* ckeditor5-font/theme/fontsize.css */
.ck-content .text-big {
    font-size: 1.4em;
}
/* ckeditor5-font/theme/fontsize.css */
.ck-content .text-huge {
    font-size: 1.8em;
}
/* ckeditor5-block-quote/theme/blockquote.css */
.ck-content blockquote {
    overflow: hidden;
    padding-right: 1.5em;
    padding-left: 1.5em;
    margin-left: 0;
    margin-right: 0;
    font-style: italic;
    border-left: solid 5px hsl(0, 0%, 80%);
}
/* ckeditor5-block-quote/theme/blockquote.css */
.ck-content[dir="rtl"] blockquote {
    border-left: 0;
    border-right: solid 5px hsl(0, 0%, 80%);
}
/* ckeditor5-basic-styles/theme/code.css */
.ck-content code {
    background-color: hsla(0, 0%, 78%, 0.3);
    padding: .15em;
    border-radius: 2px;
}
/* ckeditor5-table/theme/table.css */
.ck-content .table {
    margin: 1em auto;
    display: table;
}
/* ckeditor5-table/theme/table.css */
.ck-content .table table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    height: 100%;
    border: 1px double hsl(0, 0%, 70%);
}
/* ckeditor5-table/theme/table.css */
.ck-content .table table td,
.ck-content .table table th {
    min-width: 2em;
    padding: .4em;
    border: 1px solid hsl(0, 0%, 75%);
}
/* ckeditor5-table/theme/table.css */
.ck-content .table table th {
    font-weight: bold;
    background: hsla(0, 0%, 0%, 5%);
}
/* ckeditor5-table/theme/table.css */
.ck-content[dir="rtl"] .table th {
    text-align: right;
}
/* ckeditor5-table/theme/table.css */
.ck-content[dir="ltr"] .table th {
    text-align: left;
}
/* ckeditor5-page-break/theme/pagebreak.css */
.ck-content .page-break {
    position: relative;
    clear: both;
    padding: 5px 0;
    display: flex;
    align-items: center;
    justify-content: center;
}
/* ckeditor5-page-break/theme/pagebreak.css */
.ck-content .page-break::after {
    content: '';
    position: absolute;
    border-bottom: 2px dashed hsl(0, 0%, 77%);
    width: 100%;
}
/* ckeditor5-page-break/theme/pagebreak.css */
.ck-content .page-break__label {
    position: relative;
    z-index: 1;
    padding: .3em .6em;
    display: block;
    text-transform: uppercase;
    border: 1px solid hsl(0, 0%, 77%);
    border-radius: 2px;
    font-family: Helvetica, Arial, Tahoma, Verdana, Sans-Serif;
    font-size: 0.75em;
    font-weight: bold;
    color: hsl(0, 0%, 20%);
    background: hsl(0, 0%, 100%);
    box-shadow: 2px 2px 1px hsla(0, 0%, 0%, 0.15);
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}
/* ckeditor5-media-embed/theme/mediaembed.css */
.ck-content .media {
    clear: both;
    margin: 1em 0;
    display: block;
    min-width: 15em;
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list {
    list-style: none;
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list li {
    margin-bottom: 5px;
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list li .todo-list {
    margin-top: 5px;
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list .todo-list__label > input {
    -webkit-appearance: none;
    display: inline-block;
    position: relative;
    width: var(--ck-todo-list-checkmark-size);
    height: var(--ck-todo-list-checkmark-size);
    vertical-align: middle;
    border: 0;
    left: -25px;
    margin-right: -15px;
    right: 0;
    margin-left: 0;
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list .todo-list__label > input::before {
    display: block;
    position: absolute;
    box-sizing: border-box;
    content: '';
    width: 100%;
    height: 100%;
    border: 1px solid hsl(0, 0%, 20%);
    border-radius: 2px;
    transition: 250ms ease-in-out box-shadow, 250ms ease-in-out background, 250ms ease-in-out border;
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list .todo-list__label > input::after {
    display: block;
    position: absolute;
    box-sizing: content-box;
    pointer-events: none;
    content: '';
    left: calc( var(--ck-todo-list-checkmark-size) / 3 );
    top: calc( var(--ck-todo-list-checkmark-size) / 5.3 );
    width: calc( var(--ck-todo-list-checkmark-size) / 5.3 );
    height: calc( var(--ck-todo-list-checkmark-size) / 2.6 );
    border-style: solid;
    border-color: transparent;
    border-width: 0 calc( var(--ck-todo-list-checkmark-size) / 8 ) calc( var(--ck-todo-list-checkmark-size) / 8 ) 0;
    transform: rotate(45deg);
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list .todo-list__label > input[checked]::before {
    background: hsl(126, 64%, 41%);
    border-color: hsl(126, 64%, 41%);
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list .todo-list__label > input[checked]::after {
    border-color: hsl(0, 0%, 100%);
}
/* ckeditor5-list/theme/todolist.css */
.ck-content .todo-list .todo-list__label .todo-list__label__description {
    vertical-align: middle;
}
/* ckeditor5-html-embed/theme/htmlembed.css */
.ck-content .raw-html-embed {
    margin: 1em auto;
    min-width: 15em;
    font-style: normal;
}
/* ckeditor5-horizontal-line/theme/horizontalline.css */
.ck-content hr {
    margin: 15px 0;
    height: 4px;
    background: hsl(0, 0%, 87%);
    border: 0;
}
/* ckeditor5-code-block/theme/codeblock.css */
.ck-content pre {
    padding: 1em;
    color: hsl(0, 0%, 20.8%);
    background: hsla(0, 0%, 78%, 0.3);
    border: 1px solid hsl(0, 0%, 77%);
    border-radius: 2px;
    text-align: left;
    direction: ltr;
    tab-size: 4;
    white-space: pre-wrap;
    font-style: normal;
    min-width: 200px;
}
/* ckeditor5-code-block/theme/codeblock.css */
.ck-content pre code {
    background: unset;
    padding: 0;
    border-radius: 0;
}
/* ckeditor5-mention/theme/mention.css */
.ck-content .mention {
    background: var(--ck-color-mention-background);
    color: var(--ck-color-mention-text);
}
@media print {
    /* ckeditor5-page-break/theme/pagebreak.css */
    .ck-content .page-break {
        padding: 0;
    }
    /* ckeditor5-page-break/theme/pagebreak.css */
    .ck-content .page-break::after {
        display: none;
    }
}
    body {
      font-family: 'Noto Sans JP', sans-serif;
      
    }
  </style>
</head>
<body class="ck-content">
<h2>This is a nice blog,</h2><figure class=\"image\"><img src=\"https://elevatetiweb.perf.target.com/elevatewebtest/Atul-Jalan.png\"><figcaption><span class=\"text-huge\" style=\"font-family:Tahoma, Geneva, sans-serif;\">KISMET</span></figcaption></figure><p style=\"text-align:justify;\">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce ut aliquet purus. Sed a dapibus augue. Vivamus pretium, arcu quis aliquam feugiat, eros justo gravida metus, eu porttitor tellus tortor vel mi. Donec in nibh nec leo gravida sodales. Nunc finibus a nulla non iaculis. Pellentesque at lobortis libero. In malesuada porta sem, a imperdiet augue molestie eu. Curabitur venenatis, risus vel suscipit cursus, neque leo facilisis massa, non dignissim est diam sit amet sapien. Morbi feugiat auctor augue nec pellentesque. Donec tempus posuere erat. Suspendisse fermentum sodales nunc vel tempor.</p><p style=\"text-align:justify;\">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ullamcorper purus id tortor placerat, ut pulvinar leo tempus. Integer ligula tellus, cursus sed diam efficitur, efficitur sagittis libero. Aliquam erat volutpat. Fusce quis tortor orci. In vestibulum, sapien a mollis ultricies, mi ligula aliquam neque, nec dictum risus velit in nunc. Vivamus ipsum nunc, mattis sit amet porttitor id, ultrices ac enim. Donec quis tellus pharetra, sagittis dolor eget, venenatis magna. Interdum et malesuada fames ac ante ipsum primis in faucibus. Etiam metus urna, iaculis id imperdiet sit amet, tincidunt id lorem. Donec vestibulum lobortis consequat. Nam ut erat at nibh rutrum imperdiet sit amet id ipsum.</p><p style=\"text-align:justify;\">Aenean iaculis finibus euismod. Pellentesque fermentum pretium nisl, eget eleifend tellus pellentesque vitae. Aliquam eleifend, erat suscipit auctor imperdiet, urna libero pretium mi, at viverra dolor lorem ut sapien. Donec ac sapien vitae mi aliquam varius. Maecenas nec dui vel magna malesuada iaculis. Suspendisse eu lacinia tortor, quis imperdiet ante. Sed porttitor semper sapien sit amet luctus. Quisque nec pretium ligula, in vehicula nulla. Vestibulum eu lacus nec libero condimentum lacinia. Nulla viverra ex nec ultricies rhoncus. Aenean nec nunc in odio elementum ullamcorper quis nec est. Donec mollis, elit id laoreet imperdiet, lacus tortor tincidunt dolor, consequat imperdiet velit mauris eu odio. Mauris blandit rhoncus venenatis. Pellentesque facilisis sapien vel faucibus condimentum.</p><p style=\"text-align:justify;\">Nam eu odio vitae eros efficitur pharetra. Nullam finibus eros sed tortor scelerisque, ut sollicitudin elit semper. Quisque non congue nulla, ac aliquam dolor. Sed a mollis ligula, eget imperdiet justo. Donec in tellus nec nisi hendrerit maximus. Suspendisse potenti. Sed vitae dapibus justo, eu luctus tellus. Mauris tempor dui in vehicula consequat. Etiam pharetra aliquet sapien, quis sodales lacus venenatis a. Nullam vestibulum arcu ligula, at euismod mauris tempus eu. In ut nisl nec elit malesuada suscipit et eget orci.</p><p style=\"text-align:justify;\">Aliquam sed ipsum tincidunt erat pulvinar egestas id porttitor est. Suspendisse nec lorem euismod, condimentum urna pretium, auctor diam. Vivamus suscipit purus eu mauris finibus tempor. Praesent suscipit vel odio vulputate dictum. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. In sodales augue sollicitudin, tempus nunc in, rhoncus justo. Donec ante eros, lacinia vel tempus sit amet, commodo ut magna. Sed sed volutpat nibh. Aliquam facilisis egestas nunc, in eleifend turpis finibus quis.</p><p style=\"text-align:justify;\">Ut at iaculis dui. Etiam diam nisl, fringilla non turpis non, efficitur dapibus leo. Etiam a tortor in arcu maximus molestie. Phasellus in faucibus nisi. Morbi aliquam mi nulla, et vehicula ante aliquam non. Ut volutpat odio in urna tristique molestie. Phasellus suscipit at diam at sollicitudin. Sed eu justo dolor. Aliquam dignissim quam elit, id eleifend leo faucibus id. Phasellus euismod placerat vulputate. Proin vulputate pellentesque nulla id sodales. Donec auctor id neque ac molestie. Morbi euismod fringilla erat efficitur mollis. Cras non tellus risus. In convallis ultricies tortor posuere blandit.</p><p style=\"text-align:justify;\">Etiam sit amet euismod tellus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Pellentesque dapibus mauris eu mi vehicula porta. Curabitur efficitur aliquam commodo. Mauris nec purus purus. Fusce mollis turpis eget dignissim vehicula. Nam eget neque ut urna sagittis vestibulum suscipit at arcu. Quisque tincidunt velit lobortis dapibus ultricies. Phasellus et nisl convallis, congue mi sed, sagittis nulla. Nam vel lacinia nibh. Integer eget nunc id nisi molestie facilisis. Maecenas efficitur feugiat elementum. Sed eget nisl id nisi ultrices rutrum. Ut euismod metus et urna ultrices interdum.</p><p style=\"text-align:justify;\">Pellentesque vehicula nisl non mollis pretium. Donec lobortis, velit et blandit tristique, ex purus scelerisque velit, eget hendrerit turpis purus sit amet mi. Duis ultrices tortor ac placerat auctor. Morbi vel interdum ligula. Sed eget purus tellus. Curabitur et lorem lobortis, fermentum est ac, ultricies nisl. Vestibulum eget tellus enim.</p><p style=\"text-align:justify;\">Suspendisse vel dignissim ante, sed pretium lorem. Nunc non lectus viverra, vulputate sapien sed, finibus ligula. Vivamus porttitor, erat nec suscipit sodales, nisi mi vestibulum turpis, non hendrerit elit lorem ut dolor. Quisque consectetur ante sit amet leo rutrum, non malesuada magna maximus. Vivamus cursus leo quam, facilisis dignissim mi imperdiet ut. Nam porta porta nisi, ac feugiat sapien mollis in. Nullam vitae quam sed purus tempus facilisis. Pellentesque nec enim eget nibh euismod laoreet sit amet ac sapien. Morbi quis mi dui.</p><p style=\"text-align:justify;\">Mauris feugiat tempus fringilla. Proin non auctor lorem. Sed et urna maximus, consequat elit vel, facilisis leo. Suspendisse facilisis vitae dolor at feugiat. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Sed tempor neque nisl, sed pellentesque enim ullamcorper quis. Mauris semper dolor nisi, at gravida nibh cursus eget. Fusce eu luctus est, in luctus enim. Nullam rhoncus felis a ullamcorper suscipit. Integer hendrerit ex eget dui laoreet, rutrum facilisis justo finibus. Nunc tincidunt, ex eget ullamcorper tincidunt, lectus massa maximus ex, in hendrerit risus libero vehicula neque. Maecenas varius sit amet augue sed iaculis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Nulla laoreet diam non ultrices fermentum. Ut non eleifend nunc. Mauris in dui venenatis nibh pellentesque mollis quis ac tortor.</p><p style=\"text-align:justify;\">Quisque lobortis sed odio porttitor ornare. Pellentesque facilisis accumsan tellus, nec iaculis sapien auctor eleifend. Sed arcu arcu, laoreet sit amet accumsan et, molestie vitae augue. Nullam venenatis nisi erat, eu gravida purus hendrerit hendrerit. Duis sapien nibh, varius non urna quis, tempus rhoncus nulla. In cursus fermentum est, tristique ultrices mauris facilisis et. Morbi tempus blandit dui, vel suscipit ipsum facilisis vel. Donec quis ligula nisl. Quisque eget odio non elit volutpat interdum ut at dolor.</p><p style=\"text-align:justify;\">Aliquam erat volutpat. Curabitur pulvinar neque dolor, ut dictum lectus dapibus quis. Maecenas lacinia, ante at iaculis pretium, augue arcu maximus lorem, id ultrices neque leo nec arcu. Nam scelerisque elit ut eros mollis, sit amet rhoncus nibh scelerisque. Pellentesque faucibus, arcu vel hendrerit laoreet, enim tellus tempor ipsum, ut efficitur mauris urna id mi. Vestibulum sit amet nisi ac mauris placerat consectetur. Fusce id ipsum felis. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Fusce vitae egestas libero. Quisque in faucibus enim. Proin rhoncus in tortor ut maximus. Suspendisse interdum malesuada purus, nec sagittis massa iaculis quis. Morbi pretium sollicitudin mauris sed pellentesque. Sed non justo et nisi tempor auctor. Mauris convallis, ex vel blandit maximus, neque sem tempus felis, non malesuada nunc nibh eu purus. Sed consectetur neque massa, eget placerat lectus finibus eu.</p><p style=\"text-align:justify;\">In imperdiet arcu ut ipsum mollis feugiat. Morbi sed mauris lacus. Aliquam vel vehicula dui. Donec semper quam a ornare mattis. Cras id libero ac nibh ornare rutrum quis sit amet sapien. Donec eu enim elit. Sed varius mattis odio, non porta risus placerat eget. Suspendisse placerat, nulla molestie molestie dignissim, sapien arcu tristique enim, a porttitor urna felis vel odio. Suspendisse laoreet ipsum ut quam euismod congue. Suspendisse in mauris mollis, ullamcorper mi in, placerat velit. Sed vitae mauris feugiat, elementum sapien efficitur, tempor odio. Mauris sollicitudin blandit massa vitae scelerisque. Nam pellentesque hendrerit facilisis. Suspendisse a dui dui. Morbi semper congue gravida.</p><p style=\"text-align:justify;\">Praesent vel blandit urna. Maecenas sed elementum tellus. Aliquam vel libero non tortor pellentesque aliquet. Nulla enim lorem, aliquet a quam vel, eleifend malesuada nibh. Donec vitae risus eu lectus dignissim vulputate a id ligula. Sed id mauris arcu. Praesent facilisis nisl ante. Cras feugiat scelerisque dolor vitae auctor. Donec vel magna ornare, dignissim diam in, dignissim ex. Mauris egestas interdum diam.</p><p style=\"text-align:justify;\">Nulla ut ex malesuada, elementum ligula non, vulputate velit. Mauris non condimentum enim, non ultricies enim. Ut tristique volutpat dapibus. Donec nulla purus, dictum id nulla a, gravida pellentesque justo. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Praesent et ante leo. Vivamus quis ligula ac justo vestibulum placerat. Curabitur sit amet faucibus lorem, nec placerat ligula. Nam id justo ut sapien vestibulum imperdiet. Integer aliquet elit risus, sed dapibus urna auctor nec. Sed ultricies turpis tincidunt augue vestibulum scelerisque. Donec a ultricies est. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nulla facilisi.</p><p>&nbsp;</p>
</body>
</html>
```

I know that this is not ideal to cache and hence I'm using networkOnly FetchPolic. But I assume that this is still somehow passed to the cache. Is there any way to completely ignore cache while making a Query request?

**Comment by micimize** [comment]
@aseef17 the fact that the string is html shouldn't really be an issue – but, are you ever fetching with `FetchPolicy.cacheOnly`? If not, it is a separate issue. To debug I need more precise context, like the graphql operation, the full data, and what exactly you do on the client before executing that produces the issue.

You can also use `FetchPolicy.noCache` to avoid the cache completely.

**Comment by micimize** [comment]
@b99andla after #795, this is one of the only issues I consider blocking the release of a stable `v4`, but I really need more info to figure out what the problem is.

**Comment by aseef17** [comment]
> @aseef17 the fact that the string is html shouldn't really be an issue – but, are you ever fetching with `FetchPolicy.cacheOnly`? If not, it is a separate issue. To debug I need more precise context, like the graphql operation, the full data, and what exactly you do on the client before executing that produces the issue.
> 
> You can also use `FetchPolicy.noCache` to avoid the cache completely.

Hey, sorry for the late response.
This issue seemed to be more towards using multiple query widgets in the widget tree which call the same graphQL query.
I wanted this to separate out what will be refetched and what will remain static but this causes the cache Miss Exception so i just reverted back to using one query widget in the tree.

**Comment by Zony-Zhao** [comment]
@aseef17 I don't know the theory behind the scene. But I did made two same query in the tree, and change one to ` fetchPolicy: FetchPolicy.noCache,` solved this issue.

**Comment by leonschwanitz** [comment]
I have found the issue. When you add a field to the top query in this example viewer, its working. Add __typename to viewer. It seems that the cache not working when there is only a list field.

**Comment by AndreHaueisen** [comment]
I have the same issue for the following query:
```
query {
    currentUser {
      attributes { 
        ... on Contact {
          id
          externalId
          relatedProfiles {
            title
            profile {
              ... on Contact {
                id
                firstName
                lastName
              }
              ... on Student {
                id
                firstName
                lastName
              }
            }
          }
        }
      }
    }
  }
```

`Policy: FetchPolicy.cacheAndNetwork`

![Screen Shot 2021-12-27 at 14 36 18](https://user-images.githubusercontent.com/13854934/147494830-cb694616-0f6e-4218-b535-8466e3d695e0.png)

**Comment by micimize** [comment]
@AndreHaueisen your query structure doesn't match that of the data (no `attribute` field, unqueried fields returned) which means your server is non-conformant.

As most comments here seem to be unrelated to cache only I'll close this

**Comment by AndreHaueisen** [comment]
Sorry, could you be a bit more specific? I don't know what you mean by "your server is non-conformant"
